addons/account/account_invoice.py

```diff
@@ -1224,6 +1224,16 @@ def _compute_price(self):
         if self.invoice_id:
             self.price_subtotal = self.invoice_id.currency_id.round(self.price_subtotal)
 
+    @api.one
+    @api.depends('product_id', 'uos_id', 'quantity')
+    def _compute_measure(self):
+        ProductUom = self.env['product.uom']
+        default_uom = self.product_id.uom_id and self.product_id.uom_id.id
+        total_qty = ProductUom._compute_qty(self.uos_id.id, self.quantity, default_uom)
+        self.th_weight = total_qty * self.product_id.weight
```

Comment:
can not use self.write({'th_weight': ..., 'real_weight': ..., 'th_volume': ...}) ?